engine/src/last_turn.rs

```diff
@@ -0,0 +1,16 @@
+use crate::position::Position;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum LastTurn {
```

Comment:
are spawns not considered here? also, why not just use a struct like this to represent turns everywhere?

Reply:
Spawns are moves here. I think I know what you mean, let's hash it out in chat and when we agree on something I will put up another PR